### PR TITLE
[RENTAL]: added 3 layer for rental

### PR DESCRIPTION
**Pull request by keda-vlad**
Implement rental list & detail endpoint and implement the return rental functionality.

### CODE REVIEW

File: src/main/java/com/team01/carsharingapp/dto/rental/CreateRentalRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.team01.carsharingapp.dto.rental;
+
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import java.time.LocalDate;
+
+public record CreateRentalRequestDto(
+        @NotBlank LocalDate returnDate,
+        @NotNull Long carId
```
Comment (Someboty) [inline src/main/java/com/team01/carsharingapp/dto/rental/CreateRentalRequestDto.java]: Add message

File: src/main/java/com/team01/carsharingapp/service/impl/RentalServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    }
+
+    private Rental createRental(CreateRentalRequestDto requestDto, User user) {
+        if (requestDto.returnDate().isBefore(LocalDate.now())) {
+            throw new RentalException("Invalid return date = " + requestDto.returnDate());
+        }
+        Car car = carRepository.findById(requestDto.carId()).orElseThrow(() ->
+                new EntityNotFoundException("Can't find car by id = " + requestDto.carId()));
+        if (car.getAmountAvailable() < 1) {
+            throw new RentalException("No free cars by car id = " + requestDto.carId());
```
Comment (Someboty) [inline src/main/java/com/team01/carsharingapp/service/impl/RentalServiceImpl.java:106]:
```suggestion
            throw new RentalException("No free cars by id = " + requestDto.carId());
```

**Review by Someboty — COMMENTED** [review]
GJ! Left few comments

**Review by Someboty — APPROVED** [review]

File: src/main/java/com/team01/carsharingapp/service/impl/RentalServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+            }
+        } else {
+            if (userId == null || userId.equals(user.getId())) {
+                rentals = rentalRepository
+                        .findAllByUserIdAndStatus(user.getId(), isActive, pageable);
+            } else {
+                throw new RentalException("User role hasn't opportunity for this");
+            }
+        }
+        return rentals.stream().map(rentalMapper::toDto).toList();
```
Comment (Denis-Balako) [inline src/main/java/com/team01/carsharingapp/service/impl/RentalServiceImpl.java:85]: Format streams according to standard rules

**Review by Denis-Balako — APPROVED** [review]
LGTM!

**Pull request merged**